components/hierarchical-view/hierarchical-view-mixin.js

```diff
@@ -0,0 +1,450 @@
+import { findComposedAncestor, isComposedAncestor } from '../../helpers/dom.js';
+import { getNextFocusable, getPreviousFocusable } from '../../helpers/focus.js';
+import { css } from 'lit-element/lit-element.js';
+import ResizeObserver from 'resize-observer-polyfill/dist/ResizeObserver.es.js';
+
+const __nativeFocus = document.createElement('div').focus;
+const escapeKeyCode = 27;
+
+export const HierarchicalViewMixin = superclass => class extends superclass {
+
+	static get properties() {
+		return {
+			shown: { type: Boolean, reflect: true },
+			childView: { type: Boolean, reflect: true, attribute: 'child-view' },
+			hierarchicalView: { type: Boolean }
+		};
+	}
+
+	static get styles() {
+		return css`
+			:host {
+				box-sizing: border-box;
+				display: none;
+				position: relative;
+				left: 0;
+				overflow: hidden;
+				width: 100%;
+				--d2l-hierarchical-view-height-transition: height 300ms linear;
```

Comment:
These height animation likes look a little odd... specifically the `const` in the  two lines below.  It looks like the height animation is not working quite right as a result.  Changing the `const` to `var` appears to fix and animates as expected... from what I see so far.